src/Link.ts

```diff
@@ -14,16 +14,21 @@ export interface LinkProperties extends VirtualDomProperties {
 	to: string;
 }
 
-const getProperties = (router: Router<any>, properties: any): LinkProperties => {
+const getProperties = (router: Router<any>, properties: any): Partial<LinkProperties> => {
 	const { to, isOutlet = true, params = {}, onClick, ...props } = properties;
-	const href = isOutlet ? router.link(to, { ...router.getCurrentParams(), ...params }) : to;
+	let href: string | undefined;
+	try {
```

Comment:
Do you think that `link` should be graceful rather than doing it in here? It could return undefined?

Reply:
There are some tests asserting that `link` throws an error in this case. It seemed to be an intentional design decision and I was hesitant to change the behavior of that API vs. just handling the error here. If it should return undefined instead I can certainly update that instead of this.

Reply:
Initially that would have been by design, but I think the use-cases of routing have changed some what since the initial design think we can up the `link` function in this instance.

Reply:
Ok, I've updated `Router` and `Route` to return `string | undefined` from `link` rather than throwing an error if a link cannot be generated. I just propagated the `undefined`s through wherever these were being used. Maybe we need to handle that case separately, but as the errors were currently not being handled I wasn't sure where/how we might want to deal with the undefined values.